sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java

```diff
@@ -485,7 +486,15 @@
     .setOptimizedBestValue(true)
     .create();
 
+  /**
+   * @deprecated since 5.2 use {@link MutableTestPlan}
+   */
+  @Deprecated
   public static final String TEST_DATA_KEY = "test_data";
+  /**
+   * @deprecated since 5.2 use {@link MutableTestPlan}
+   */
+  @Deprecated
   public static final Metric<String> TEST_DATA = new Metric.Builder(TEST_DATA_KEY, "Unit tests details", Metric.ValueType.DATA)
```

Comment:
:new: ![INFO](https://raw.githubusercontent.com/henryju/image-hosting/master/info.png)Do not forget to remove this deprecated code someday.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)